packages/core/src/Step.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import Flex from './Flex'
+import Text from './Text'
+import Icon from 'pcln-icons'
+
+const Container = styled(Flex)`
+  cursor: ${props => (props.onClick ? 'pointer' : 'default')};
+`
+
+function Step({ className, isCurrent, isCompleted, children, onClick }) {
+  const color = isCurrent || isCompleted ? 'primary' : 'text.light'
+
+  return (
+    <Container className={className} alignItems="center" onClick={onClick}>
+      {isCompleted && (
+        <Icon name="check" color="primary" size={16} mr={1} legacy={false} />
+      )}
+      <Text color={color} bold={isCurrent} fontSize={1}>
+        {children}
+      </Text>
+    </Container>
+  )
+}
+
+Step.displayName = 'Step'
+
+Step.propTypes = {
+  className: PropTypes.string,
+  isCurrent: PropTypes.bool,
+  isCompleted: PropTypes.bool,
```

Comment:
Secondly, I think these names could be `active` and `completed`. Reasons:

- adjective prop names could omit "is" prefix, as they imply boolean values
- current step in some designs may not be the active step word that needs bold style. I think `active` as a prop name seems more expressive to avoid ambiguity. (see inspiration from the [Step API of Material-UI](https://material-ui.com/api/step/#props)

P.S.: I also think `disabled ` prop is worth of consideration for the `Step` implementation, to allow designs to render a step UI that's just for visual purpose, without any functionalities. You can consider to add `disabled` prop in this PR or log an issue to follow up.

Reply:
I'll add an issue as I need this released asap.